perfkitbenchmarker/configs/storage_utility.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python
+
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import numpy
+
+
+class SQLProviderError(Exception):
+  pass
+
+
+class IOPSError(Exception):
+  pass
+
+
+class StorageConfig(object):
+  """Given iops and service provider requirements, return necessary parameters.
+
+  Assumes SSD persistent disks.
+
+  Args:
+    iops: Number of iops required.
+    provider: Provider: 'AWS' or 'GCP'.
+  """
+
+  AWS = 'AWS'
+  GCP = 'GCP'
+
+  def __init__(self, iops, provider=GCP):
+    self._storage_size = None
+    self._cpu_count = None
+    self._number_disks = None
+    self._iops = iops
+    self._provider = provider.upper()
+    self.ValidateProvider()
+    self.ValidateIOPS()
+    self.PopulateConfigs()
+
+  def ValidateProvider(self):
+    """Validate provider to be GCP or AWS, throw exception if invalid.
+
+    Raises:
+      SQLProviderError: Incorrect provider type given.
+    """
+    if self._provider != self.AWS and self._provider != self.GCP:
+      raise SQLProviderError('Provider given is not supported by '
+                             'storage_utility.')
+
+  def ValidateIOPS(self):
+    """Validate IOPS to be positive integer, throw exception if invalid.
+
+    Raises:
+      IOPSError: Invalid IOPS parameter given.
+    """
+    if self._iops < 1:
+      raise IOPSError()
+
+  def PopulateConfigs(self):
+    """Poplulate Storage Configurations."""
+    self.SetStorageSize()
+    self.SetCPUCount()
+    self.SetNumberDisks()
+
+  def PrintConfigs(self):
+    """Print out necessary configs."""
+    vm_config = ("For {} IOPS using {}, the following is required:\n\tStorage "
+                 "Size (GB): {}\n\tCPU Count: {}\n\tNumber of "
+                 "Disks: {}").format(self._iops,
+                                     self._provider.upper(),
+                                     self._storage_size,
+                                     self._cpu_count,
+                                     self._number_disks)
+    print vm_config
+
+  def SetStorageSize(self):
```

Comment:
Clarify here that the size we determine is the minimum size required to achieve the specified IOPS.